### DISCUSSION TITLE
Deploying RMM by using GPO

### DISCUSSION THREAD

**Discussion by Matchag** [amidaware/tacticalrmm#732] [Show and tell]
Hey there,

I just deployed the agent **by using GPO**.
I had few problems so I'm creating a discussion to help people who whant to do the same thing.

First of all, I tried to deploy it just with a **.bat script which execute the installer** (it didn't work), then I tried with the **winagent.exe and the .bat script** who @wh1te909 made. Unfortunately, it didn't work (I set up the GPO at computer's boot), then I tried to launch the script at **sessions's opening** which kinda worked. In fact, it worked, but it asked the AD's admin credentials. 

Then, I tried the install with the .ps1 script, which didn't work at all (I think the Policy with PowerShell script was Restricted that's why it didn't worked). 

I retried with the **.bat with the winagent.exe**, but this time at the **shutdown** of the PC. It worked !
It wasn't really clean, so I **edited the .bat** by adding **timeout 10 /nobreak** at the beginning of the script. 
**It perfectly worked**.

I'm just a beginner in this domain so I might have missed something, but I think it can help if I share you my experiences with TRMM.

I'm happy if this helped you !

**Comment by dinger1986** [comment]
Thanks for the write up.

It should really have worked a number of ways but glad you found a solution that worked.

**Comment by bbrendon** [comment]
Can't you just call the exe directly as a logon script?

**Reply by Matchag** [reply]
I did, but it asked for the admin's rights

**Reply by dinger1986** [reply]
use Group Policy not login scripts

**Comment by bradhawkins85** [comment]
Use the PowerShell script to deploy via GPO, run it as a startup script.

I've written the script to wait for network connectivity before starting the install but also not hang the startup process too long if it can't connect to the RMM server.

It's how I deploy in all my domain environments and it's been set and forget, apart from updating the script occasionally to make sure to deploy a current version of the agent.

Just set a long expiry when you generate the PowerShell script.

**Comment by SrLiath** [comment]
I tried to use a PS1 script to deploy RMM, but it didn’t work well. I have some machines running Windows 7 and Windows XP (yeah, I know, but I can't update them), so I did this:
https://github.com/SrLiath/TaticalRmmMassiveInstallAd

It's more like creating an MSI installer to deploy over my network via GPO, without user interaction, and using a VBS script for older systems, is simple, but works.

**Reply by wh1te909** [reply]
nice, thanks! added it to our awesome repo: https://github.com/amidaware/trmm-awesome?tab=readme-ov-file#install-tactical-agent-via-msi--active-directory

**Reply by P6g9YHK6** [reply]
waitaminute?? is this capable of fixing the issue linked to old versions of windows ? what kind of witchcraft has been used ?

**Reply by SrLiath** [reply]
Haha, not exactly. I just used the same MSI to install our custom client on XP. It works fine on Windows 7 (almost fully on 64-bit), but now I see there's an issue installing it on 32-bit. I'll try a different approach now.

**Reply by P6g9YHK6** [reply]
ok i see you made me think that you had found some kind of magic stone that could solve the TLS issues
